src/native/fs/fs_napi.cpp

```diff
@@ -1597,12 +1600,15 @@ struct LinkFileAt : public FSWrapWorker<FileWrap>
     }
     virtual void Work()
     {
+        int fd = _wrap->_fd;
+        CHECK_WRAP_FD(fd);
+
         // gpfs_linkat() is the same as Linux linkat() but we need a new function because
         // Linux will fail the linkat() if the file already exist and we want to replace it if it existed.
-        if (_replace_fd == 0) {
-            SYSCALL_OR_RETURN(dlsym_gpfs_linkat(_wrap->_fd, "", AT_FDCWD, _filepath.c_str(), AT_EMPTY_PATH));
+        if (_replace_fd == -1) {
+            SYSCALL_OR_RETURN(dlsym_gpfs_linkat(fd, "", AT_FDCWD, _filepath.c_str(), AT_EMPTY_PATH));
```

Comment:
is it really valid to use fd=-1 in this call ?? seems like an error unless we explain why it's correct

Reply:
sorry my bad - this fd is not the same as _replace_fd. resolving this comment.